Scripts/Source/minai_Sex.psc

```diff
@@ -1359,3 +1307,79 @@ int function GetGender(actor targetActor)
 
   return sex
 endfunction
+
+function onSexStart(int ThreadID, string framework)
+  bool playerInvolved = isPlayerInvolved(ThreadID, framework)
+
+  if playerInvolved
+    AIFF.ChillOut()
+  EndIf
+
+  ; enable sex-mode for php
+  SetSexSceneState("on")
+  UpdateThreadTable("startthread", framework, ThreadID)
+
+  ; get actors to store random actor who will talk at the end. Doing it at start ensure that on sex end events we won't end up where in threads there are no actors anymore
+  Actor[] actors
+  if(framework == ostimType)
+    actors = OThread.GetActors(ThreadID)
+  elseif(framework == sexlabType)
+    actors = slf.GetController(ThreadID).Positions
+  endif
+
+  JMap.setForm(actorToSayOnEndMap, ThreadID, getWeightedRandomActorToSpeak(actors))
+endfunction
+
+function onSexEnd(int ThreadID, string framework)
+  ; disable sex-mode for php. AIFF implementation is a little incorrect, since there are still
+  SetSexSceneState("off")
+  UpdateThreadTable("end", framework, ThreadID)
+
+  ; get actor to say something after sex
+  bool playerInvolved = isPlayerInvolved(ThreadID, framework)
+  actor actorToSayOnEnd = JMap.getForm(actorToSayOnEndMap, ThreadID) as Actor
+  if bHasAIFF
+    AIFF.SetAnimationBusy(0, Main.GetActorName(actorToSayOnEnd))
+  EndIf
+  sexTalkOnEnd(actorToSayOnEnd, playerInvolved, framework)
+  ; clean actor to say from finished thread
+  JMap.removeKey(actorToSayOnEndMap, ThreadID)
+endfunction
+
+function onSceneChange(int ThreadID, string framework)
+  UpdateThreadTable("scenechange", sexlabType, ThreadID)
+  bool playerInvolved = isPlayerInvolved(ThreadID, framework)
+  if(playerInvolved)
+    AIFF.ChillOut()
+  endif
+
+  actor[] actors
+  if(framework == ostimType)
+    actors = OThread.GetActors(ThreadID)
+  elseif(framework == sexlabType)
+    actors = slf.GetController(ThreadID).Positions
+  endif
+  ; do we really need this check for sexlab?
+  ; if (sortedActorList.length < 1)
+  ;   return
+  ; EndIf
+
+  ; we don't to call VR here, right? We still log VR events in context, sexTalk is triggered by events
```

Comment:
question about VR, it sends events into context which appear later in llm context, I don't think we need call vr sextalk separately. I commented for now, but if needed can remove it completely

Reply:
Agreed - I think let's keep CBPC for out of sex dialogue with your new system. I reenabled it for that after your last update, and made it not fire in sex.